include/mav/Message.h

```diff
@@ -457,7 +505,7 @@ namespace mav {
 
             header().magic() = 0xFD;
             header().len() = payload_size;
-            header().incompatFlags() = 0;
+            header().incompatFlags() = sign ? 0x01 : 0x00;
```

Comment:
As mentioned before, giving a `sign` boolean to `finalize` has the user expecting that the message will be signed. However, that's not the case, as you have to call `sign` to actually sign it. So can you change it that either `finalize` does everything needed for signing when requested to sign?

Reply:
Yeah, I'll quote my question from an earlier comment here

> Also, if we're doing the signing in finalize(), I'd need to pass both the timestamp and key in as arguments, which seems like a rather cumbersome amount of args for one function. 

Is this your request?